### DISCUSSION TITLE
Welcome to TinyTuya Discussions!

### DISCUSSION THREAD

**Discussion by jasonacox** [jasonacox/tinytuya#77] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us how you use TinyTuya and you interest areas with home automation, etc.

**Comment by Dev-crunch** [comment]
TinyTuya is only correctly reading the status of Tuya plug correctly on the first status read, subsequent reads are the same despite the plug status including current, voltage, & power changing. Output:

Python 3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> 
========= RESTART: C:/Users/Appa/Documents/Python/Local Tuya test 7.py =========
Before OutletDevice
Before Set Version
Before Status
data_dict={'dps': {'1': False, '9': 0, '18': 0, '19': 0, '20': 2370, '21': 1, '22': 557, '23': 25743, '24': 14191, '25': 2940, '26': 0, '38': 'memory', '39': False, '42': '', '43': ''}}
{'1': False, '9': 0, '18': 0, '19': 0, '20': 2370, '21': 1, '22': 557, '23': 25743, '24': 14191, '25': 2940, '26': 0, '38': 'memory', '39': False, '42': '', '43': ''}
0
0.0
237.0
ON ^^^^^^^^^^^^^^^^^^^^^
30/10/2022 21:47:24- current=0mA, voltage=237.0V, power=0.0W
OFF _________________
current=0mA, voltage=237.0V, power=0.0W
ON ^^^^^^^^^^^^^^^^^^^^^
30/10/2022 21:47:45- current=0mA, voltage=237.0V, power=0.0W
OFF _________________
current=0mA, voltage=237.0V, power=0.0W
ON ^^^^^^^^^^^^^^^^^^^^^
30/10/2022 21:48:06- current=0mA, voltage=237.0V, power=0.0W
OFF _________________
current=0mA, voltage=237.0V, power=0.0W
ON ^^^^^^^^^^^^^^^^^^^^^
30/10/2022 21:48:27- current=0mA, voltage=237.0V, power=0.0W
OFF _________________
current=0mA, voltage=237.0V, power=0.0W
ON ^^^^^^^^^^^^^^^^^^^^^
30/10/2022 21:48:49- current=0mA, voltage=237.0V, power=0.0W
OFF _________________
current=0mA, voltage=237.0V, power=0.0W
ON ^^^^^^^^^^^^^^^^^^^^^
30/10/2022 21:49:10- current=0mA, voltage=237.0V, power=0.0W
OFF _________________

The simple program : 
#***************************************************************************
#
# 29/10/2022 DS Appears to be working in terms of correctly picking up current, voltage.
# and power when the program is first run. But it is holding those values without
# updating them when they change!
#
# 30/10/2022 DS Cleaned up from earlier test version.
#***************************************************************************



# Example Usage of TinyTuya
import tinytuya
import time
import json
from datetime import datetime



#print("Before set debug")   
#tinytuya.set_debug(True)  #For raw messages

#***************************************
#Initialisation
#***************************************
print("Before OutletDevice")   
d = tinytuya.OutletDevice('removed', 'removed', 'removed')

print("Before Set Version")  
d.set_version(3.3) #3.3 works

#***************************
#Get Status
#***************************
print("Before Status")
d.set_dpsUsed({"1": None})  # This needs to be a datapoint available on the device
data_dict = d.status()              #Python dictionary object returned
print('data_dict=%r' %data_dict)    

print(data_dict.get("dps","unknown")) #Get contents of parent "dps" return unknow if not found
data_dict_dps=data_dict.get("dps","unknown")
print(data_dict_dps.get("18","unknown")) #Get value of key 18, current [mA]
print(data_dict_dps.get("19","unknown")/10) #Get value of key 19, power [W]
print(data_dict_dps.get("20","unknown")/10) #Get value of key 20, voltage [V]


#*******************************
# Continuously read A,W,V
#*******************************
i=5
while i<100:
    
 d.turn_on()
 print( "ON ^^^^^^^^^^^^^^^^^^^^^")
 time.sleep(5)
 data_dict = d.status()
 now_obj=datetime.now()
 dt_string=now_obj.strftime("%d/%m/%Y %H:%M:%S")
 current=data_dict_dps.get("18","unknown")
 power=data_dict_dps.get("19","unknown")/10
 voltage=data_dict_dps.get("20","unknown")/10
 print (dt_string+"- current="+str(current)+"mA, voltage="+str(voltage)+"V, power="+str(power)+"W")
 time.sleep(5)
 
 d.turn_off()
 print ("OFF _________________")
 time.sleep(5)
 data_dict = d.status()              
 current=data_dict_dps.get("18","unknown")
 power=data_dict_dps.get("19","unknown")/10
 voltage=data_dict_dps.get("20","unknown")/10
 print ("current="+str(current)+"mA, voltage="+str(voltage)+"V, power="+str(power)+"W")
 time.sleep(5)
 
 i +=1

print("After while loop")

**Reply by uzlonewolf** [reply]
`d.updatedps( [18, 19, 20] )` before `d.status()`

**Reply by Dev-crunch** [reply]
Good idea. I found the bug in my program after trying your suggestion. I have a logical error.  I will neverthless try your command on the next iteration of the program. Thanks

**Comment by FrBerger83** [comment]
Hello,

I am working on a home integration of tuya devices .
I successfully accessed my device from home computer, but would like to be able to run the status queries from an ESP device.
I installed micropython on an ESP8266, but doesn't knox how to install tinytuya on the ESP.
is it possible, is it a stupid question?
Thanks for the work done and thanks in advance for a response.

**Reply by uzlonewolf** [reply]
I was able to hack a bunch of stuff out and get it kinda-sorta working with the PC port of micropython, but haven't spent the time to make it truly compatible.  See #421

**Reply by FrBerger83** [reply]
Thanks for the response
